src/vmm/src/device_manager/mmio.rs

```diff
@@ -100,18 +100,20 @@ pub struct MMIODeviceInfo {
 /// Manages the complexities of registering a MMIO device.
 pub struct MMIODeviceManager {
     pub(crate) bus: devices::Bus,
-    mmio_base: u64,
-    next_avail_mmio: u64,
+    pub(crate) address_allocator: AddressAllocator,
     pub(crate) legacy_irq_allocator: IdAllocator,
     pub(crate) id_to_dev_info: HashMap<(DeviceType, String), MMIODeviceInfo>,
 }
 
 impl MMIODeviceManager {
     /// Create a new DeviceManager handling mmio devices (virtio net, block).
     pub fn new(mmio_base: u64, irq_interval: (u32, u32)) -> MMIODeviceManager {
+        #[cfg(target_arch = "x86_64")]
+        let size = 768 << 20;
+        #[cfg(target_arch = "aarch64")]
+        let size = 1 << 30; //>> 1GB
```

Comment:
AFAICT this is the length of the MMIO gap, right? We shouldn't be using magic constants
For x86 we should reuse the `MEM_32BIT_GAP_SIZE` variable, while for aarch64 I believe this is `DRAM_MEM_START - MAPPED_IO_START`, right?

Maybe we should also have the size as a parameter, to try to keep this code more arch-agnostic